kf_lib_data_ingest/target_api_plugins/kids_first_dataservice.py

```diff
@@ -986,8 +987,14 @@ def submit(host, entity_class, body):
         resp = _POST(host, api_path, body)
 
     if resp.status_code in {200, 201}:
-        result = resp.json()["results"]
-        return result["kf_id"]
+        result = resp.json()
+        if isinstance(result, dict) and "results" in result:
+            return result["results"].get("kf_id")
+        else:
+            raise RequestException(
+                f"Got response code 200/201 but response is not as expected."
+                f"Response was:\n{pformat(result)}"
+            )
```

Comment:
If the response is coming back in an unexpected format, we don't know that it will parse as json. What about this?
```suggestion
        try:
            return resp.json()["results"]["kf_id"]
        except Exception as e:
            raise RequestException(
                f"Got status 200/201 from /{api_path} but response was not as expected:"
                f"\nSent:\n{body}\nGot:\n{resp.text}"
           ) from e
```

Reply:
much better